libraries/chain/include/eosio/chain/thread_utils.hpp

```diff
@@ -41,14 +47,15 @@ namespace eosio { namespace chain {
       std::vector<std::thread>       _thread_pool;
       boost::asio::io_context        _ioc;
       std::optional<ioc_work_t>      _ioc_work;
+      on_except_t                    _on_except;
    };
 
 
-   // async on thread_pool and return future
+   // async on io_context and return future
    template<typename F>
-   auto async_thread_pool( boost::asio::io_context& thread_pool, F&& f ) {
+   auto async_thread_pool( boost::asio::io_context& ioc, F&& f ) {
```

Comment:
I agree, when the function was first created it did use the `thread_pool` instead of `io_context`. I'll update the name to better reflect what it does now.